src/shared/sam/debugger/awsSamDebugger.ts

```diff
@@ -26,17 +30,101 @@ export class AwsSamDebugConfigurationProvider implements vscode.DebugConfigurati
         token?: vscode.CancellationToken
     ): Promise<AwsSamDebuggerConfiguration | undefined> {
         if (!AWS_SAM_DEBUG_REQUEST_TYPES.has(debugConfiguration.request)) {
-            vscode.window.showInformationMessage('Invalid request type')
+            vscode.window.showErrorMessage('Invalid request type')
+
+            return undefined
+        }
+
+        if (!AWS_SAM_DEBUG_TARGET_TYPES.has(debugConfiguration.invokeTarget.target)) {
+            vscode.window.showErrorMessage('Invalid invokeTarget.target type')
+
+            return undefined
         }
-        if (
-            debugConfiguration.invokeTarget &&
-            debugConfiguration.invokeTarget.target &&
-            !AWS_SAM_DEBUG_TARGET_TYPES.has(debugConfiguration.invokeTarget.target)
-        ) {
-            vscode.window.showInformationMessage('Invalid invokeTarget.target type')
+
+        if (debugConfiguration.invokeTarget.target === 'template') {
+            if (!isTemplateDebugConfigValid(debugConfiguration)) {
+                return undefined
+            }
+        } else if (debugConfiguration.invokeTarget.target === 'code') {
+            if (
+                !debugConfiguration.lambda ||
+                !debugConfiguration.lambda.runtime ||
+                !samLambdaRuntimes.has(debugConfiguration.lambda.runtime)
+            ) {
+                vscode.window.showErrorMessage(
+                    'Debug config of invokeTarget.target: code must have a valid lambda.runtime'
+                )
+
+                return undefined
+            }
         }
+
         vscode.window.showInformationMessage('Not implemented')
 
-        return undefined
+        return debugConfiguration
+    }
+}
+
+function isTemplateDebugConfigValid(debugConfiguration: AwsSamDebuggerConfiguration): boolean {
+    const templateTarget = (debugConfiguration.invokeTarget as any) as AwsSamDebuggerInvokeTargetTemplateFields
+    const cftRegistry = CloudFormationTemplateRegistry.getRegistry()
+
+    const template = cftRegistry.getRegisteredTemplate(templateTarget.samTemplatePath)
+
+    if (!template) {
+        vscode.window.showErrorMessage('invokeTarget.target.samTemplatePath not found')
+
+        return false
     }
+
+    if (
+        !template.template.Resources ||
+        !Object.keys(template.template.Resources).includes(templateTarget.samTemplateResource)
+    ) {
+        vscode.window.showErrorMessage('invokeTarget.target.samTemplateResource not found in template')
+
+        return false
+    }
+
+    if (
+        template.template.Resources[templateTarget.samTemplateResource]?.Type !==
+        CloudFormation.SERVERLESS_FUNCTION_TYPE
+    ) {
```

Comment:
attn: @awschristou @justinmk3 

This check is currently irrelevant due to https://github.com/aws/aws-toolkit-vscode/blob/master/src/shared/cloudformation/cloudformation.ts#L175 ; we straight up reject any templates that don't contain a serverless function. Thinking I should remove that requirement, but just want to make sure first.

Reply:
It is best not to make assumptions about what other code is doing. The template loading code currently couples loading and validating together, which isn't ideal. If that code is remedied, upstream code that is implemented based on that behaviour may start to fail.

Reply:
So I think changing that is out of scope for this PR; I'll go ahead and leave out the test testing this specific line (since it'll never fire, unless I purpose-build a mock registry to check this) but leave the check in code. Sound good?

Reply:
Changing the template loading code is out of scope indeed.
Leaving this check in place is important, so is testing it (without going to extreme measures).

Are you able to test this by providing a template containing both a lambda and a non-lambda resource, then crafting a debug configuration that references the non-lambda resource?

Reply:
Kinda-sorta. If you have two resources with one as a SAM function and the other as something else, it'll only load the SAM function. That means you won't be testing that the other resource isn't a SAM function, you'd only be testing whether or not it exists (which is tested elsewhere).